revocation/ocsp/ocsp.go

```diff
@@ -167,17 +167,30 @@ func extensionsToMap(extensions []pkix.Extension) map[string][]byte {
 func executeOCSPCheck(cert, issuer *x509.Certificate, server string, opts Options) (*ocsp.Response, error) {
 	// TODO: Look into other alternatives for specifying the Hash
 	// https://github.com/notaryproject/notation-core-go/issues/139
-	ocspRequest, err := ocsp.CreateRequest(cert, issuer, &ocsp.RequestOptions{Hash: crypto.SHA256})
+	// The following do not support SHA256 hashes:
+	//  - Microsoft
+	//  - Entrust
+	//  - Let's Encrypt
+	//  - Digicert (sometimes)
+	// As this represents a large percentage of public CAs, we are using the
+	// hashing algorithm SHA1, which has been confirmed to be supported by all
+	// that were tested.
+	ocspRequest, err := ocsp.CreateRequest(cert, issuer, &ocsp.RequestOptions{Hash: crypto.SHA1})
 	if err != nil {
 		return nil, GenericError{Err: err}
 	}
 
 	var resp *http.Response
-	if base64.URLEncoding.EncodedLen(len(ocspRequest)) >= 255 {
+	postRequired := base64.StdEncoding.EncodedLen(len(ocspRequest)) >= 255
+	var encodedReq string
+	if !postRequired {
+		encodedReq = url.QueryEscape(base64.StdEncoding.EncodeToString(ocspRequest))
+		postRequired = len(encodedReq) >= 255
+	}
+	if postRequired {
```

Comment:
How about:
```suggestion
	postRequired := base64.StdEncoding.EncodedLen(len(ocspRequest)) >= 255
	var encodedReq string
	if !postRequired {
		encodedReq = url.QueryEscape(base64.StdEncoding.EncodeToString(ocspRequest))
		if len(encodedReq) >= 255 {
		    reader := bytes.NewReader(ocspRequest)
		    resp, err = opts.HTTPClient.Post(server, "application/ocsp-request", reader)
		}
	} else {
	    var reqURL string
	    reqURL, err = url.JoinPath(server, encodedReq)
	    ...
	}
```

Reply:
This logic isn't quite right. We want the post to happen for both situations when it is longer (which would require duplication of that code if we nest the if statements). Also, the get should be used for if it is the smallest. I think I can refactor this to try to make it cleaner

Reply:
I have edited my suggestion above.

Reply:
Updated this logic in the most recent commit